<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Settings</class>
 <widget class="QDialog" name="Settings">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>650</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>550</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>设置</string>
  </property>
  <layout class="QVBoxLayout" name="mainLayout">
   <item>
    <widget class="QGroupBox" name="generalGroup">
     <property name="title">
      <string>常规设置</string>
     </property>
     <layout class="QFormLayout" name="generalLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="labelListenAddr">
        <property name="text">
         <string>监听地址</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QVBoxLayout">
        <item>
         <widget class="QLineEdit" name="editListenAddr">
          <property name="text">
           <string>127.0.0.1:30000</string>
          </property>
          <property name="placeholderText">
           <string>SOCKS5/HTTP代理监听地址和端口</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="labelListenAddrHint">
          <property name="text">
           <string>格式: IP:端口 (如: 127.0.0.1:30000 或 0.0.0.0:1080)</string>
          </property>
          <property name="styleSheet">
           <string>color: gray; font-size: 10px;</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="labelAutoStart">
        <property name="text">
         <string>开机自启</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QCheckBox" name="checkAutoStart">
        <property name="text">
         <string>开机时自动启动</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="labelMinimizeToTray">
        <property name="text">
         <string>最小化到托盘</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QCheckBox" name="checkMinimizeToTray">
        <property name="text">
         <string>关闭窗口时最小化到系统托盘</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="tunDnsGroup">
     <property name="title">
      <string>TUN 模式 DNS</string>
     </property>
     <layout class="QFormLayout" name="tunDnsLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="labelTunnelDNS">
        <property name="text">
         <string>远端 DNS 服务器</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QVBoxLayout">
        <item>
         <widget class="QLineEdit" name="editTunnelDNS">
          <property name="text">
           <string>8.8.8.8</string>
          </property>
          <property name="placeholderText">
           <string>IPv4 DNS服务器地址</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="labelTunnelDNSHint">
          <property name="text">
           <string>⚠️ 流量自动通过代理隧道传输到远端DNS</string>
          </property>
          <property name="styleSheet">
           <string>color: #ff8c00; font-size: 10px;</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="labelTunnelDNSv6">
        <property name="text">
         <string>远端 DNS 服务器 (IPv6)</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <layout class="QVBoxLayout">
        <item>
         <widget class="QLineEdit" name="editTunnelDNSv6">
          <property name="text">
           <string>2001:4860:4860::8888</string>
          </property>
          <property name="placeholderText">
           <string>IPv6 DNS服务器地址</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="labelTunnelDNSv6Hint">
          <property name="text">
           <string>Google DNS IPv6 地址，可留空禁用IPv6 DNS</string>
          </property>
          <property name="styleSheet">
           <string>color: gray; font-size: 10px;</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="tunGroup">
     <property name="title">
      <string>TUN 模式设置</string>
     </property>
     <layout class="QFormLayout" name="tunLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="labelTunIP">
        <property name="text">
         <string>IPv4 地址</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="editTunIP">
        <property name="text">
         <string>10.0.85.2/24</string>
        </property>
        <property name="placeholderText">
         <string>CIDR 格式，如 10.0.85.2/24</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="labelTunMTU">
        <property name="text">
         <string>MTU</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="spinTunMTU">
        <property name="minimum">
         <number>1280</number>
        </property>
        <property name="maximum">
         <number>9000</number>
        </property>
        <property name="value">
         <number>1380</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="labelTunStack">
        <property name="text">
         <string>网络栈</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QComboBox" name="comboTunStack">
        <item>
         <property name="text">
          <string>system</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>gvisor</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="labelTunAutoRoute">
        <property name="text">
         <string>自动路由</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QCheckBox" name="checkTunAutoRoute">
        <property name="text">
         <string>启用自动路由配置</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="labelTunStrictRoute">
        <property name="text">
         <string>严格路由</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QCheckBox" name="checkTunStrictRoute">
        <property name="text">
         <string>启用严格路由模式</string>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Settings</receiver>
   <slot>accept()</slot>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Settings</receiver>
   <slot>reject()</slot>
  </connection>
 </connections>
</ui>
